### PR TITLE
Document using an EventSourcedBehavior as a PersistentFSM

### PR DESCRIPTION
**Pull request by chbatey**
Re-uses the code from the migration guide as the mgiration logic
is in adapters.

Almost seems very obvious you can do this but can't hurt to have.

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/persistence-fsm.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# EventSourced behaviors as finite state machines
+
+An @apidoc[EventSourcedBehavior] can be used to represet a persistent FSM. If you're migrating an existing classic
```
Comment (patriknw) [inline akka-docs/src/main/paradox/typed/persistence-fsm.md]: typo represet

**Review by patriknw — APPROVED** [review]
LGTM, one small typo

**Review by johanandren — APPROVED** [review]

File: akka-docs/src/main/paradox/typed/persistence-fsm.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# EventSourced behaviors as finite state machines
+
+An @apidoc[EventSourcedBehavior] can be used to represent a persistent FSM. If you're migrating an existing classic
+persistent FSM to EventSourcedBehavior see the @ref[migration guide](../persistence-fsm.md#migration-to-eventsourcedbehavior).
+
+To demonstrate this consider an example of a shopping application. A an customer can be in the following states:
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/persistence-fsm.md]:
```suggestion
To demonstrate this consider an example of a shopping application. A customer can be in the following states:
```

File: akka-docs/src/main/paradox/typed/persistence-fsm.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+* Looking around
+* Shopping (has something in their basket)
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/persistence-fsm.md]: Remove the qualifier `(has something in their basket)`?

File: akka-docs/src/main/paradox/typed/persistence-fsm.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+* Get current cart 
+
+Scala
+:  @@snip [PersistentFsmToTypedMigrationSpec.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/PersistentFsmToTypedMigrationSpec.scala) { #commands }
+
+Java
+:  @@snip [PersistentFsmToTypedMigrationCompileOnlyTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/PersistentFsmToTypedMigrationCompileOnlyTest.java) { #commands }
+
+The command handler of the EventSourcedBehavior is used to convert the commands that change the state of the FSM
+to events and reply to commands.
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/persistence-fsm.md]:
```suggestion
to events, and reply to commands.
```

File: akka-docs/src/main/paradox/typed/persistence-fsm.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+@scala[The command handler:]@java[The `forStateType` command handler can be used:]
+
+Scala
+:  @@snip [PersistentFsmToTypedMigrationSpec.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/PersistentFsmToTypedMigrationSpec.scala) { #command-handler }
+
+Java
+:  @@snip [PersistentFsmToTypedMigrationCompileOnlyTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/PersistentFsmToTypedMigrationCompileOnlyTest.java) { #command-handler }
+
+The event handler is used to change state once the events have been persisted. When the EventSourcedBehavior is restarted
+the events are replayed to get back into the correct state.
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/persistence-fsm.md]: Or consider
```
To return to the correct state, when the EventSourcedBehavior is restarted the events are replayed.
```

**Comment by patriknw** [comment]
Refs https://github.com/akka/akka/issues/25473

**Pull request merged**